src/pages/SignUp/SignUp.js

```diff
@@ -1,8 +1,140 @@
-import React from "react";
-import "./SignUp.scss";
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { useNavigate } from 'react-router-dom';
+
+import './SignUp.scss';
 
 function SignUp() {
-  return <div>SignUp</div>;
+  const navigate = useNavigate();
+
+  const [hasEmail, setHasEmail] = useState(false);
+  const [values, setValues] = useState({
+    first_name: '',
+    last_name: '',
+    email: '',
+    password: '',
+    cofirm_password: '',
+  });
+
+  const handleInput = e => {
+    const { name, value } = e.target;
+    setValues({ ...values, [name]: value });
+
+    if (values.email) {
+      setHasEmail(false);
+    }
+  };
+
+  const { first_name, last_name, email, password } = handleInput;
+
+  const emailRegExp =
+    /^[0-9a-zA-Z]([-_\\.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_\\.]?[0-9a-zA-Z])*\.[a-zA-Z]{2,3}$/;
+  const passwordRegExp =
+    /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$ %^&*-]).{8,}$/;
+  const cofirmEmail = values.email && !emailRegExp.test(values.email);
+  const cofirmPassword =
+    values.cofirm_password && values.password !== values.cofirm_password;
+  const cofirmRegPassword =
+    values.password && !passwordRegExp.test(values.password);
+
+  const onSubmit = async e => {
+    e.preventDefault();
+
+    if (cofirmEmail && cofirmPassword && cofirmRegPassword) {
+      const url = 'http://10.58.6.177:8000/users/signup';
+      try {
+        const response = await fetch(url, {
+          method: 'POST',
+          body: JSON.stringify({
+            first_name,
+            last_name,
+            email,
+            password,
+          }),
+        });
+
+        const result = await response.json();
+        if (result.message === 'THIS_EMAIL_ALREADY_EXISTS') {
+          setHasEmail(true);
+        }
+
+        if (result.message === 'SIGNUP_SUCCESS') {
+          navigate('/signin');
+          // navigate('/welcome');
+        }
+      } catch (err) {
+        alert(err);
+      }
+    }
+  };
+
+  return (
+    <main className="sign-in-container">
+      <div className="sign-in-inner">
+        <div className="inner-left">
+          {hasEmail ? (
+            <label className="input-label">이미 사용중인 이메일 입니다.</label>
+          ) : null}
```

Comment:
falsy할 경우가 null이라면 논리곱 단축평가를 활용하시면 될 것 같습니다.
```suggestion
          {hasEmail && <label className="input-label">이미 사용중인 이메일 입니다.</label>}
```

Reply:
다른 부분도 참고해서 수정해주세요!

Reply:
수정 완료!

Reply:
추가적으로  구조 분해 할당하여 수정해봤습니다